src/client.ts

```diff
@@ -1,124 +1,154 @@
-import http from 'http';
+import axios, { AxiosRequestConfig } from 'axios';
+import axiosRetry from 'axios-retry';
 
-import { IntegrationProviderAuthenticationError } from '@jupiterone/integration-sdk-core';
+import {
+  IntegrationProviderAuthenticationError,
+  IntegrationProviderAPIError,
+  IntegrationLogger,
+} from '@jupiterone/integration-sdk-core';
 
 import { IntegrationConfig } from './config';
-import { AcmeUser, AcmeGroup } from './types';
+import {
+  SignalSciencesUser,
+  SignalSciencesCorp,
+  SigSciResponseFormat,
+} from './types';
 
 export type ResourceIteratee<T> = (each: T) => Promise<void> | void;
 
+const BASE_URI = 'https://dashboard.signalsciences.net/api/v0';
+
 /**
- * An APIClient maintains authentication state and provides an interface to
- * third party data APIs.
- *
- * It is recommended that integrations wrap provider data APIs to provide a
- * place to handle error responses and implement common patterns for iterating
- * resources.
+ * API Client providing access to key Signal Science resources.
+ * NOTE: pagination is not supported by provider on /corps and /users.
+ * Pagination, in general, is supported by other endpoints. This will
+ * need to be considered during future development.
+ * TODO: Add pagination.
  */
-export class APIClient {
-  constructor(readonly config: IntegrationConfig) {}
+export class SignalSciencesAPIClient {
+  constructor(
+    readonly logger: IntegrationLogger,
+    readonly config: IntegrationConfig,
+  ) {
+    axiosRetry(axios, {
```

Comment:
is this configurable to only retry certain status codes? For example, `401`, `403`, `404` are usually not retryable

Reply:
Great question. The package docs specify the following `By default, it retries if it is a network error or a 5xx error on an idempotent request (GET, HEAD, OPTIONS, PUT or DELETE).`